Add Perspective (a TransformComponent)

nattridgell opened the pull request:
Key points for Perspective
- constructor
  - takes a single SimpleLength with type `px`
  - the length must be strictly positive ( > 0 )
- matrix
  - 3D identity + ( -1 / length ) at 4th row, 3rd column
- is2DComponent
  - always false (perspective is a 3D component)
- cssString is for example `perspective(10px)`

Commit:
Add Perspective (a TransformComponent)

Key points for Perspective
 - constructor
   - takes a single SimpleLength with type `px`
   - the length must be strictly positive ( > 0 )
 - matrix
   - 3D identity + ( -1 / length ) at 4th row, 3rd column
 - is2DComponent
   - always false (perspective is a 3D component)
 - cssString is for example `perspective(10px)`

wilddamon commented on src/perspective.js:
```diff
@@ -0,0 +1,52 @@
+    if (arguments.length != 1) {
+      throw new TypeError('Perspective takes exactly 1 argument.');
+      // TODO: Change check of 'px' when LengthValue.LengthType enum is updated.
+    } else if (!(length instanceof SimpleLength) || length.type != 'px') {
```
Since the later ifs can't be reached if one of the others is matched, these can be

``` javascript
if (condition) {
  throw ...
}
if (condition) {
  throw
}
```

This is enforced by the style guide in C++ so may as well get used to it now :)

nattridgell replied:
Thanks for the note :-) Will do.

Commit:
Change style of if statements for errors

 - do not use else if statements, just a series of if statements
   for multiple error checks in constructors.
 - add TODO to store a copy of the length given in Perspective
   once LengthValue copy constructors are implemented.

wilddamon commented on src/perspective.js:
```diff
@@ -0,0 +1,55 @@
+    if (arguments.length != 1) {
+      throw new TypeError('Perspective takes exactly 1 argument.');
+    }
+    // TODO: Change check of 'px' when LengthValue.LengthType enum is updated.
```
This has been landed now.

Commit:
Merge branch 'master' into transformPerspective

wilddamon commented:
LGTM
